docs/source/cc_service.md

```diff
@@ -0,0 +1,276 @@
+```
+
+## Configuring a peer to process external chaincode
+
+If you have reviewed the external build and chaincode launcher documentation then these steps will be familiar to you. We leverage those scripts to define the external chaincode service information. These scripts are located on the peer file system that is accessible to the peer process from the `core.yaml` file in the `externalBuilders` element under the `chaincode` stanza. An example is included in the steps which follow.
+
+### Create the set of external builder and launcher scripts on the peer
+
+In order to configure chaincode as an external service, you must use the scripts as follows:
+* **detect** - examines the chaincode package and accept if metadata.json `type` is set to `external`.
+* **build** - builds the chaincode and places the build artifacts in the `BUILD_OUTPUT_DIR`. For chaincode as an external service, the peer does not build the chaincode. Instead, the script extracts the chaincode endpoint information in the `connection.json` file and other artifacts from the `code.tar.gz` and places them in the specified location.
+* **release** - copies the built artifacts (in our case the `connection.json` file) to a specified release location.
+
+You may notice that the external builder and launcher `bin/run` script is not required for the chaincode as an external service.
+
+The scripts that are required to exist in the peer `/bin` directory:
+```
+    <fully qualified path on the peer's env>
+    └── bin
+        ├── build
+        ├── detect
+        └── release
+```
+
+### Modify the peer core.yaml to include the externalBuilder
+
+Finally, in order for the peer to use the external builder and launcher scripts, you need to modify the `chaincode` stanza of the peer `core.yaml` file to include the `externalBuilders` configuration element.
+
+```yaml
+externalBuilders:
+     - name: myexternal
+       path: <fully qualified path on the peer's env>   
+```
+
+## External builder and launcher sample scripts
+
+To help understand what each script needs to contain to work with the chaincode as an external service, this section contains samples of  `bin/detect` `bin/build`, `/bin/release`, and `bin/run` scripts.
+
+**Note:** These samples use the `jq` command to parse json. You can run `jq --version` to check if you have it installed. Otherwise, install `jq` or suitably modify the scripts yourself.
+
+### /bin/detect
+
+The `bin/detect script` is responsible for determining whether or not a buildpack should be used to build a chaincode package and launch it. For chaincode as an external service, the script should detect that metadata.json `type` is set to `external`.  The peer invokes detect with two arguments:
+
+```
+bin/detect CHAINCODE_SOURCE_DIR CHAINCODE_METADATA_DIR
+```
+
+A typical `detect` script could contain:
+
+```sh
+
+#!/bin/bash
+
+set -euo pipefail
+
+if [ "$#" -ne 2 ]; then
+    >&2 echo "Expected 2 directories got $#"
+    exit 2
+fi
```

Comment:
We're needlessly complicating examples by having these assertions. It looks like we're just copying the contents of tests.